### DISCUSSION TITLE
Git cannot clone submodule drjit

### DISCUSSION THREAD

**Discussion by yangocean-sudo** [mitsuba-renderer/mitsuba3#1210] [Q&A] [answered]
Hi, I'm using Linux and try to git clone the repository:
` git clone --recursive https://github.com/mitsuba-renderer/mitsuba3`
After that, I ran:
`git submodule update --init --recursive`
to ensure that all submodules are correctly initialized.
However, I met an error:
fatal: reference is not a tree: 42b27b4343b4ae7c1a4df75ba54f4ab7bc7b3fd1
Unable to checkout '42b27b4343b4ae7c1a4df75ba54f4ab7bc7b3fd1' in submodule path 'ext/drjit'.
I've tried following command to solve this problem:
`
deinit -f ext/drjit 
git rm -f ext/drjit
rm -rf .git/modules/ext/drjit 
git submodule add https://github.com/mitsuba-renderer/drjit.git ext/drjit  
git submodule update --init --recursive
`
But when I type `pip install .` in folder mitusba3, I faced an error:
 `  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
INFO: pip is looking at multiple versions of mitsuba to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement drjit==0.4.2 (from mitsuba) (from versions: 0.4.4, 0.4.5, 0.4.6)
ERROR: No matching distribution found for drjit==0.4.2`
What should I do to solve this problem? Many thanks.

**Comment by merlinND** [comment, marked as answer]
Hello @yangocean-sudo,

The `master` branch is undergoing a lot of development at the moment and can be unstable. I would recommend checking out the last tagged version and trying to pull again:

```bash
git clone --recursive https://github.com/mitsuba-renderer/mitsuba3 --branch v3.5.2
```

Or, keeping your existing clone, something like (untested):

```bash
cd mitsuba3
git checkout v3.5.2
git submodule sync
git submodule update --init --recursive
```
